quelpa.el

```diff
@@ -1606,13 +1624,27 @@ Return t in each case."
 
 (defun quelpa-update-cache (cache-item)
   ;; try removing existing recipes by name
-  (setq quelpa-cache (cl-remove (car cache-item)
-                                quelpa-cache :key #'car))
+  (setq quelpa-cache (cl-remove-if
+                      (lambda (item) (eq (car cache-item) (car item)))
+                      quelpa-cache))
   (push cache-item quelpa-cache)
   (setq quelpa-cache
         (cl-sort quelpa-cache #'string<
                  :key (lambda (item) (symbol-name (car item))))))
 
+(defun quelpa-remove-cache (cache-item)
+  "Remove cache-item from `quelpa-cache'."
+  (let ((cache-version (plist-get (cdr cache-item) :version)))
+    (setq quelpa-cache
+          (cl-remove-if
+           (lambda (item)
+             (and (eq (car cache-item) (car item))
```

Comment:
It may be better to use `equal` rather than `eq` here.  Generally `equal` is more likely to be correct.

Reply:
We're comparing symbol here so `equal` is a little bit overkill IMO. We've already using `eq` quite a lot in this file. Maybe we can refactoring this file later to switch to `equal`?
Same thing for doc string too.

Reply:
AFAIK `eq` is fine for comparing symbols, so as long as that's always what's being compared, it should be fine.  I'll defer to your judgement here.

In general, using `equal` will always work properly, and using it by default helps prevent future bugs. AFAIK `eq` should only be used if necessary.